index.js

```diff
@@ -40,14 +42,24 @@ var assertionVisitor = {
 				return;
 			}
 
+			var hoisted = hoistAwaitAndYield(path, arg0);
+
 			path.node.arguments[0] = wrapWithHelper({
 				HELPER_ID: t.identifier(this.avaThrowHelper()),
-				EXP: arg0,
+				EXP: hoisted ? hoisted.arg0 : arg0,
 				LINE: t.numericLiteral(arg0.loc.start.line),
 				COLUMN: t.numericLiteral(arg0.loc.start.column),
 				SOURCE: t.stringLiteral(state.file.code.substring(arg0.start, arg0.end)),
 				FILE: t.stringLiteral(state.file.opts.filename)
 			}).expression;
+
+			if (hoisted) {
+				path.replaceWithMultiple(
+					hoisted.expressions.concat(
+						t.expressionStatement(path.node)
```

Comment:
I haven't had the time to get back to this. We could explore whether we can run this plugin _before_ any other presets and plugins. Alternatively the generator compilation (for Node.js v4 and up) should reduce to `yield` expressions.